frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/Node.java

```diff
@@ -40,6 +40,29 @@ public Node(String host, Set<String> labels, ResourceDescriptor totalResource, R
     this.requestedResource = ResourceDescriptor.newInstance(0, 0, 0, 0L);
   }
 
+  // Compare two node's AvailableResource,  order is Gpu, Cpu, Memory
+  @Override
+  public int compareTo(Node arg0) {
+    if (this.getAvailableResource().getGpuNumber() > arg0.getAvailableResource().getGpuNumber())
+      return 1;
+    if (this.getAvailableResource().getGpuNumber() < arg0.getAvailableResource().getGpuNumber()) {
+      return -1;
```

Comment:
Suggest to implement compare inside ResourceDescriptor

Reply:
This compare is only compare gpu, cpu and memory. not that general to apply to all ResourceDescription. 

---
In reply to: [174681105](https://github.com/Microsoft/pai/pull/318#discussion_r174681105) [](ancestors = 174681105)